src/Console/ConfigurationResolver.php

```diff
@@ -147,14 +147,17 @@ final class ConfigurationResolver
      */
     private ?iterable $finder = null;
 
-    private $format;
+    private ?string $format = null;
 
     /**
      * @var null|Linter
      */
     private $linter;
 
-    private $path;
+    /**
+     * @var null|list<string>
```

Comment:
why `list` and not `iterable` or `array` ?
(not challenging, i honestly want to understand the difference)

Reply:
especially when getter is returning `string[]`

Reply:
This property is only used as an array internally, there is no point in widening the type with `iterable`.

`list<T>` is same as `array<int, T>` with two additional constraints that make it more relevant IMO:
* keys start at 0;
* keys are consecutive with no gaps.

Maybe the return type of the getter could be changed to `list<string>` as well, but that's not the scope of this PR.

Reply:
is there any difference between `list<T>` and `T[]` ?

Reply:
I think `iterable` is smaller type of `array`. for me it's very similar as saying we expose `FooInterface` rather than `Foo` (concrete implementation)

Reply:
I think `T[]` is strictly equivalent to `array<int|string, T>`.

Reply:
> I think `iterable` is smaller type of `array`. for me it's very similar as saying we expose `FooInterface` rather than `Foo` (concrete implementation)

I agree but here the property is only used internally, having the most specific type is better IMO. If we used an object instead, would you introduce an interface just for the sake of having it as the property's type?

Reply:
If internal usage, probably "no". but as soon as it's available in public interface of (even internal) class - "yes".

here, we are lucky that we don't introduce it, we can use what's in place already (`iterable` already exists, not that we need to create it)

Reply:
aso @julienfalque to be clear, my intention was to understand the difference (`(not challenging, i honestly want to understand the difference)`), I'm not asking to blindly change it nor blocking this PR in current shape!

Whole PR brings nice value for sure!